src/ripple/app/tx/impl/Transactor.cpp

```diff
@@ -854,6 +854,12 @@ Transactor::operator()()
     }
 #endif
 
+    if (auto trap = ctx_.app.trapTxID();
+        trap && ctx_.tx.getTransactionID() == *trap)
+    {
+        JLOG(j_.debug()) << "Transaction trapped: " << *trap;
+    }
```

Comment:
After you change the function to retun a `const std::optional<uint256>&` you can just wrtite

    if(ctx_.app.trapTxID() == ctx.tx.getTransactionID())
        JLOG(j_.debug()) << "Transaction trapped: " << ctx.tx.getTransactionID();

Reply:
On second thought, I think it's not such a good idea, as it will mean unconditional copying of `uint256` from `ctx.tx` for the purpose of equality comparison. I've changed it back again to enable bool shortcircuit evaluation, where in normal production use we only check if the trap is set and that's all. Of course that means more branching, but in normal production use we have one branch anyway.

Reply:
I  looked at this more carefully because no copying should happen given overload 21 on https://en.cppreference.com/w/cpp/utility/optional/operator_cmp but then I looked closer and noticed that `STTx::getTransactionID` returns a copy 🤦  

I agree that because of that it will copy and I suggest you fix that code to return a `const uint256&` instead!

Reply:
Thank you for the suggestion, however such a change does not belong to this PR.